pyqtorch/core/operation.py

```diff
@@ -13,17 +13,39 @@
 # limitations under the License.
 
 from typing import Any
-
+from functools import partial
 import torch
 from numpy.typing import ArrayLike
 
 from pyqtorch.converters.store_ops import ops_cache, store_operation
 from pyqtorch.core.utils import _apply_gate
 
+
 IMAT = torch.eye(2, dtype=torch.cdouble)
 XMAT = torch.tensor([[0, 1], [1, 0]], dtype=torch.cdouble)
 YMAT = torch.tensor([[0, -1j], [1j, 0]], dtype=torch.cdouble)
 ZMAT = torch.tensor([[1, 0], [0, -1]], dtype=torch.cdouble)
+SMAT = torch.tensor([[1, 0], [0, 1j]], dtype=torch.cdouble)
+TMAT = torch.tensor([[1, 0], [0, torch.exp(torch.tensor(1j) * torch.pi / 4)]], dtype=torch.cdouble)
+SWAPMAT = torch.tensor([[1, 0, 0, 0], [0, 0, 1, 0], [0, 1, 0, 0], [0, 0, 0, 1]], dtype=torch.cdouble)
+NOTMAT = XMAT
+
+
+def get_parametrized_matrix_for_operation(operation_type: str, theta: torch.Tensor) -> torch.Tensor:
+    def pass_param_to_parametrized_matrix(theta: torch.Tensor, matrix: torch.Tensor) -> torch.Tensor:
+        return IMAT * torch.cos(theta / 2) - 1j * matrix * torch.sin(theta / 2)
+    if operation_type == "RX":
+        return pass_param_to_parametrized_matrix(theta, XMAT)
+    elif operation_type == "RY":
+        return pass_param_to_parametrized_matrix(theta, YMAT)
+    elif operation_type == "RZ":
+        return pass_param_to_parametrized_matrix(theta, ZMAT)
```

Comment:
@dominikandreasseitz Maybe you can just build a dictionary for this?

Reply:
yeah, good idea

Reply:
done

Reply:
i refactored the code to use a global variable OPERATIONS_DICT which gets imported from utils. this way, you dont have to change it in both batched_operation and operation.